src/openrct2/peep/peep.c

```diff
@@ -8321,36 +8341,56 @@ static sint32 peep_interact_with_path(rct_peep* peep, sint16 x, sint16 y, rct_ma
 
 		uint8 rideIndex = map_element->properties.path.ride_index;
 
+		// Check if this queue path tile is connected to a ride.
 		if (rideIndex == 0xFF){
+			// Queue is not connected to a ride.
 			peep->interactionRideIndex = 0xFF;
 			return peep_footpath_move_forward(peep, x, y, map_element, vandalism_present);
 		}
 
+		// Queue is connected to a ride.
 		if (peep->state == PEEP_STATE_QUEUING){
+			// Check if this queue is connected to the ride the
+			// peep is queuing for, i.e. the player hasn't edited
+			// the queue.
 			if (peep->current_ride == rideIndex){
 				return peep_footpath_move_forward(peep, x, y, map_element, vandalism_present);
 			}
+			// Queue got disconnected from the ride.
 			remove_peep_from_queue(peep);
 			peep_decrement_num_riders(peep);
 			peep->state = PEEP_STATE_1;
 			peep_window_state_update(peep);
 			return peep_footpath_move_forward(peep, x, y, map_element, vandalism_present);
 		}
 
+		// One of two cases applies here:
+		// 1. Peep is in the ride queue but no longer queuing.
+		//    This usually means the ride was closed.
+		// 2. Peep is walking up to the queue entrance from
+		//    a normal path tile.
+
 		if (peep->interactionRideIndex == rideIndex){
+			// Case 1 above applies, so walk the queue.
 			return peep_footpath_move_forward(peep, x, y, map_element, vandalism_present);
 		}
 
+		// Case 2 above applies - decide whether to go on the ride.
 		peep->var_F4 = 0;
 		uint8 stationNum = (map_element->properties.path.additions & 0x70) >> 4;
 		if (!peep_should_go_on_ride(peep, rideIndex, stationNum, PEEP_RIDE_DECISION_AT_QUEUE)){
-			peep->interactionRideIndex = rideIndex;
+			// Peep has decided not to go on the ride.
```

Comment:
Correct me if wrong but I thought the reason it set the interactionRideIndex was to prevent it from considering going on this ride again until it had tried a different ride. So really we should change Case 1 to check that it has been added to the queue rather than change case 2.